code/game/machinery/doppler_array.dm

```diff
@@ -88,4 +155,53 @@ var/list/doppler_arrays = list()
 			stat &= ~NOPOWER
 		else
 			icon_state = "[initial(icon_state)]-off"
-			stat |= NOPOWER
\ No newline at end of file
+			stat |= NOPOWER
+
+/obj/machinery/doppler_array/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "doppler_array.tmpl", "Tachyon-doppler array", 500, 650)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/doppler_array/ui_data(mob/user, ui_key = "main", datum/topic_state/state = default_state)
+	var/data[0]
+	var/list/explosion_data = list()
+	for(var/datum/explosion_log/E in logged_explosions)
+		explosion_data += list(list(
+			"logged_time" = E.logged_time,
+			"epicenter" = E.epicenter,
+			"actual_size_message" = E.actual_size_message,
+			"theoretical_size_message" = E.theoretical_size_message,
+			"unique_datum_id" = E.unique_datum_id))

```

Comment:
Why call `UID()` on its own in the datum's `New()`, rather than here?

Reply:
I'll move it here, I think I just put it in `New` because I didn't consider it didn't need to be called until the point in which they access the nanoUI menu.